File: Divide-a-String-Into-Groups-of-Size-k.java
class Solution {
    public String[] divideString(String s, int k, char fill) {
        String[] parts=new String[s.length()/k + (s.length()%k==0 ? 0:1)];
        for(int i=0;i<s.length();i+=k){
            parts[i/k]=s.substring(i,Math.min(i+k,s.length()));
        }
        StringBuilder str= new StringBuilder(parts[parts.length-1]);
        while(str.length()<k){
            str.append(fill);
        }
        parts[parts.length-1]=str.toString();
        return parts;
    }
}

File: Divide-a-String-Into-Groups-of-Size-k.cpp
class Solution {
public:
    vector<string> divideString(string s, int k, char fill) {
        vector<string> parts;
        for(int i=0;i<s.size();i+=k){
            parts.push_back(s.substr(i,k));
        }
        while(parts[parts.size()-1].size()<k){
            parts[parts.size()-1].push_back(fill);
        }
        return parts;
    }
};